packages/core/src/services/chains/ProviderProcessor/saveOrPublishProviderLogs.ts

```diff
@@ -29,8 +31,14 @@ export async function saveOrPublishProviderLogs<T extends StreamType>({
     data: { ...data, streamType } as AIProviderCallCompletedData<T>,
   })
 
+  const workspace = await unsafelyFindWorkspace(data.workspaceId)
+  if (!workspace) {
+    throw new NotFoundError('Workspace not found')
+  }
```

Comment:
Wait. No need for this. Pass `workspace` from `run.ts` This code is only used there:
https://github.com/latitude-dev/latitude-llm/blob/c8f02d0da03376a1fc5adc7adc662964dc2f8a6e/packages/core/src/services/chains/run.ts#L137